### ISSUE TITLE
I am trying to get response from spring oauth2 api

### ISSUE THREAD

**Issue by wilbard** [axios/axios#2996]
**Section/Content To Improve**
Quote or link to section

**Suggested Improvement**
Identify what is confusing or incorrect and what could make it better

**Relevant File(s)**: [e.g. README.md]

**Comment by wilbard** [comment]
**This is the request:**
getProduct = async () => {
return await axios.get("http://stock.cibo.co.tz:8844/product/view/product?product_id=455");
}
**and this is the response:**
Error: Request aborted

**But when the request is inside set timeout:**
getProduct = async () => {
return await setTimeout( () => { axios.get("http://stock.cibo.co.tz:8844/product/view/product?product_id=455") }, 0);
}
**It will return 2, in which the real response can only be seen on the console requests details**

**Comment by carloschida** [comment]
I don't think this is an axios issue but rather an Access-Control headers issue.

In this [sandbox](https://codesandbox.io/s/romantic-microservice-obgvz?file=/src/App.js), my browser reports that the 'XMLHttpRequest cannot load http://stock.cibo.co.tz:8844/product/view/product?product_id=455 due to access control checks.'

On the other hand, running this snippet on node could help you debug your issue a bit more independently form the browser environment:
```js
const axios = require("axios");

const getProduct = async () => {
	return await axios.get("http://stock.cibo.co.tz:8844/product/view/product?product_id=455");
}

getProduct().then(console.log)

const getProduct2 = async () => {
	return setTimeout(() => {
		axios.get("http://stock.cibo.co.tz:8844/product/view/product?product_id=455")
	}, 0);
}

getProduct2().then(console.log)
```

Here, `getProduct` will print the request response, and the second will time out.

**Comment by wilbard** [comment]
> I don't think this is an axios issue but rather an Access-Control headers issue.
> 
> In this [sandbox](https://codesandbox.io/s/romantic-microservice-obgvz?file=/src/App.js), my browser reports that the 'XMLHttpRequest cannot load http://stock.cibo.co.tz:8844/product/view/product?product_id=455 due to access control checks.'
> 
> On the other hand, running this snippet on node could help you debug your issue a bit more independently form the browser environment:
> 
> ```js
> const axios = require("axios");
> 
> const getProduct = async () => {
> 	return await axios.get("http://stock.cibo.co.tz:8844/product/view/product?product_id=455");
> }
> 
> getProduct().then(console.log)
> 
> const getProduct2 = async () => {
> 	return setTimeout(() => {
> 		axios.get("http://stock.cibo.co.tz:8844/product/view/product?product_id=455")
> 	}, 0);
> }
> 
> getProduct2().then(console.log)
> ```
> 
> Here, `getProduct` will print the request response, and the second will time out.

Thank you Carlos, I tried to read various examples then I came to realize that I must put the axios request  or method containing axios inside componentDidMount() rather than just puting it on any method like how I used to code on angular. Now its working, thank you

**Comment by carloschida** [comment]
@wilbard I'm glad you found the root cause :)
Please, consider closing this issue to reduce the clutter.
